apps/backend/src/services/zfs/dataset_service.py

```
@@ -0,0 +1,145 @@
+                "properties": properties,
+                "timestamp": datetime.now(timezone.utc).isoformat()
+            }
+            
+        except Exception as e:
+            self.logger.error(f"Error getting dataset properties for {dataset_name} on {hostname}: {e}")
+            raise ZFSError(f"Failed to get dataset properties: {str(e)}", operation="get_dataset_properties", hostname=hostname)
+
+    async def create_dataset(self, hostname: str, dataset_name: str, properties: Optional[Dict[str, str]] = None, timeout: int = 60) -> Dict[str, Any]:
+        """Create a new ZFS dataset"""
+        try:
+            cmd = f"zfs create"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)